logistics_project/apps/malawi/warehouse/runner.py

```diff
@@ -541,8 +586,12 @@ def _update_reporting_rate(supply_point, report_period, products_managed, base_l
     late_cutoff = report_period.window_date + \
         timedelta(days=settings.LOGISTICS_DAYS_UNTIL_LATE_PRODUCT_REPORT)
 
+    # Filtering on base_level is not necessary for ProductReport because the supply_point
+    # should tell what the base_level is (base_level will be HSA if the supply_point
+    # is an hsa, and base_level will be FACILITY if the supply_point is a facility).
+    # So since this is already a big query, it's better to not include the filter
+    # for performance.
     reports_in_range = ProductReport.objects.filter(
-        product__type__base_level=base_level,
```

Comment:
@czue see my comment above - i thought about this a bit but let me know if you disagree